src/liblelantus/lelantus_primitives.h

```diff
@@ -27,6 +28,8 @@ struct NthPower {
 
     void go_next() {
         pow *= num;
+        if (pow == Scalar(uint64_t(1)))
+            throw std::invalid_argument("NthPower resulted 1");
```

Comment:
I think this has DoS potential because `NthPower` is used in muptiple places and not only with challenge value. I.e. attacker can set some value in proof to 1 and if the exception is not properly caught it'll crash all the nodes. It's better to move the check to higher level.

Reply:
I have put this call in try-catch in verifier side. I think that is enoygh for preventing such risks.

Reply:
will doublecheck to verify that I did not missed it somewhere.